packages/console/src/widget.ts

```diff
@@ -274,6 +274,13 @@ class CodeConsole extends Widget {
     super.dispose();
   }
 
+  /**
+   * Toggle ForeignHandler
+   */
+  toggleForeignHandler() {
+    this._foreignHandler.enabled = ! this._foreignHandler.enabled;
```

Comment:
Instead of a toggle, can we just expose this boolean attribute using a get/set pair? Maybe it could be named `.echoForeignData` or simply just `.echo` - not sure about a good name?

I think the toggling logic should be in the command, and at this level, it makes more sense to have the strictly more general get/set pair creating a new attribute exposing the enabled attribute.